autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/deduper/AutofillDeduplicationUsernameAndPasswordMatcher.kt

```diff
@@ -17,19 +17,22 @@
 package com.duckduckgo.autofill.impl.deduper
 
 import com.duckduckgo.autofill.api.domain.app.LoginCredentials
+import com.duckduckgo.autofill.impl.username.AutofillUsernameComparer
 import com.duckduckgo.di.scopes.AppScope
 import com.squareup.anvil.annotations.ContributesBinding
 import javax.inject.Inject
 
 interface AutofillDeduplicationUsernameAndPasswordMatcher {
 
-    fun groupDuplicateCredentials(logins: List<LoginCredentials>): Map<Pair<String?, String?>, List<LoginCredentials>>
+    suspend fun groupDuplicateCredentials(logins: List<LoginCredentials>): Map<Pair<String?, String?>, List<LoginCredentials>>
 }
 
 @ContributesBinding(AppScope::class)
-class RealAutofillDeduplicationUsernameAndPasswordMatcher @Inject constructor() : AutofillDeduplicationUsernameAndPasswordMatcher {
+class RealAutofillDeduplicationUsernameAndPasswordMatcher @Inject constructor(
+    private val usernameComparer: AutofillUsernameComparer,
+) : AutofillDeduplicationUsernameAndPasswordMatcher {
 
-    override fun groupDuplicateCredentials(logins: List<LoginCredentials>): Map<Pair<String?, String?>, List<LoginCredentials>> {
-        return logins.groupBy { it.username to it.password }
+    override suspend fun groupDuplicateCredentials(logins: List<LoginCredentials>): Map<Pair<String?, String?>, List<LoginCredentials>> {
+        return usernameComparer.groupUsernamesAndPasswords(logins)
```

Comment:
use `usernameComparer` instead of its own grouping logic where it would not consider case sensitivity of username